File: 0217_contains_duplicate.py
# https://leetcode-cn.com/problems/contains-duplicate/

class Solution:
    def containsDuplicate(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: bool
        """

        count = {}
        for i in nums:
            count[i] = count.get(i,0) + 1
            if count[i] > 1:
                return True
        return False

File: Cpp/0217_contains_duplicate.cpp
// https://leetcode.cn/problems/contains-duplicate/

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> hash_table;
        for (int num: nums) {
            if (hash_table.find(num) != hash_table.end()) {
                return true;
            }
            hash_table[num] = 1;
        }
        return false;
    }
};
